pandas/tests/copy_view/test_methods.py

```diff
@@ -516,16 +517,16 @@ def test_shift_columns(using_copy_on_write):
     df2 = df.shift(periods=1, axis=1)
 
     assert np.shares_memory(get_array(df2, "2020-01-02"), get_array(df, "2020-01-01"))
-    df.iloc[0, 1] = 0
+    df.iloc[0, 0] = 0
     if using_copy_on_write:
         assert not np.shares_memory(
             get_array(df2, "2020-01-02"), get_array(df, "2020-01-01")
         )
-    expected = DataFrame(
-        [[np.nan, 1], [np.nan, 3], [np.nan, 5]],
-        columns=date_range("2020-01-01", "2020-01-02"),
-    )
-    tm.assert_frame_equal(df2, expected)
+        expected = DataFrame(
+            [[np.nan, 1], [np.nan, 3], [np.nan, 5]],
+            columns=date_range("2020-01-01", "2020-01-02"),
+        )
+        tm.assert_frame_equal(df2, expected)
```

Comment:
Haha this was a tricky one 😀

The previous test was buggy, since we set 0 into the second column, but shift removed the second column from the DataFrame, so that is kind of pointless. It's not really related to this PR, can do in a follow-up if you prefer.

Reply:
Some additional info:

```
assert np.shares_memory(get_array(df2, "2020-01-02"), get_array(df, "2020-01-01"))
```

This basically shows what we are testing now. Non-CoW will modify both objects

Reply:
Ah, yes, I see, it's shifting the columns